### PR TITLE
get docs for stats and plots to link to ArviZ

### PR DESCRIPTION
**Pull request by OriolAbril**
As discussed in slack, this should make stats and plots api pages to link to ArviZ counterparts, the current look is this one:

image

Uses intersphinx whose configuration has been added to `conf.py`. 

Depending on what your PR does, here are a few things you might want to address in the description:
+ [x] what are the (breaking) changes that this PR makes? None
+ [x] important background, or details about the implementation
+ [x] right before it's ready to merge, mention the PR in the RELEASE-NOTES.md

### CODE REVIEW

**Comment by twiecki** [comment]
This is really cool! I'll refer to @ColCarroll who has done most on docs to confirm that it looks right. 

@OriolAbril I think you can already go ahead and add this to the release notes.

**Review by AlexAndorra — APPROVED** [review]
This 👏  is 👏  awesome 👏 , thanks a lot @OriolAbril ! I don't see any missing bits where I can help you, but tell me if there are. Waiting for Doc-Master Colin's approval to merge 😉

**Comment by OriolAbril** [comment]
Things to do would be double checking docs can be generated properly, make sure there are no more linked references to plots or stats functions that have to be updated and review the descriptions and lists of functions to make sure I have not missed any.

**Review by aloctavodia — APPROVED** [review]

**Comment by twiecki** [comment]
Thanks @OriolAbril!

**Pull request merged**
